src/liballoc/collections/btree/set.rs

```diff
@@ -1254,7 +1255,12 @@ impl<'a, T: Ord> Iterator for Intersection<'a, T> {
                     match Ord::cmp(small_next, other_next) {
                         Less => small_next = small_iter.next()?,
                         Greater => other_next = other_iter.next()?,
-                        Equal => return Some(small_next),
+                        Equal => {
+                            if small_iter.len() > other_iter.len() {
```

Comment:
Why swap them? It looks like the implementation of `next` would be correct without swapping (though we would want to change the name of `small_iter` as you observed). The reason to avoid swapping would be that this code typically runs many times while `size_hint` would typically run at most once. We could have instead `size_hint` take the minimum between the two child lengths.

Reply:
Thank you. I have changed to this way.